src/core/ext/xds/xds_client.cc

```diff
@@ -1414,6 +1414,13 @@ void XdsClient::Orphan() {
   // Clear cache and any remaining watchers that may not have been cancelled.
   authority_state_map_.clear();
   invalid_watchers_.clear();
+  // We may still be sending lingering queued load report data, so don't
+  // just clear the load reporting map, but we do want to clear the refs
+  // we're holding to the ChannelState objects, to make sure that
+  // everything shuts down properly.
+  for (auto& p : xds_load_report_server_map_) {
+    p.second.channel_state.reset(DEBUG_LOCATION, "XdsClient::Orphan()");
```

Comment:
I don't think it's possible, but just in case, I've changed the code there to check if the value is already null.  (It looks like there was already a pre-existing bug there where we'd deference the iterator if it pointed to `end()`, and I've fixed that as well -- but I suspect that if we had been hitting that code-path, we'd have already been seeing that bug anyway.)